File: Module.Persist.TPM/Model/History/HistoricalCompetitorPromo.cs
﻿using System;
using Core.History;
using Module.Persist.TPM.Model.TPM;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(CompetitorPromo))]
    public class HistoricalCompetitorPromo : BaseHistoricalEntity<Guid>
    {
        public string CompetitorName { get; set; }
        public string ClientTreeFullPathName { get; set; }
        public int ClientTreeObjectId { get; set; }
        public DateTimeOffset? StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public string Name { get; set; }
        public int Number { get; set; }
        public string MechanicType { get; set; }
        public double? Price { get; set; }
        public double? Discount { get; set; }
        public string CompetitorBrandTechBrandTech { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/competitorpromo/HistoricalCompetitorPromo.js
Ext.define('App.model.tpm.competitorpromo.HistoricalCompetitorPromo', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'CompetitorPromo',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'CompetitorName', type: 'string', isDefault: true},
        { name: 'ClientTreeFullPathName', type: 'string', isDefault: true },
        { name: 'CompetitorBrandTechName', type: 'string', isDefault: true},
        { name: 'Name', type: 'string', isDefault: true },
        { name: 'Number', type: 'int', isDefault: true },
        { name: 'StartDate', type: 'date', timeZone: +3, convert: dateConvertTimeZone },
        { name: 'EndDate', type: 'date', isDefault: false, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'Discount', type: 'float', isDefault: true },
        { name: 'Price', type: 'float', isDefault: true },
        { name: 'Subrange', type: 'string', isDefault: true },
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalCompetitorPromoes',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        },
        extraParams: {
            //Id промо для метода GetById в истории
            Id: null
        }
    }
});
